### PR TITLE
feat(imagegallerymodal): new component created to select an image

### LINKED ISSUE #1786: [Component Proposal] Image gallery modal
### What package is this for?

- [x] React
- [ ] Angular

### Summary

The image card needs the ability to search and choose from a list of photos in the Dashboard editor flow. The image gallery component will be in a Modal and should have:
- search functionality
- icon tabs that switch between gallery and list view
- checkmark selected state for list (does the gallery also contain a checkmark?)

Link to [invision](https://ibm.invisionapp.com/share/KWO0FZ5BHQ9#/screens/319748234)
[Internal issue](https://github.ibm.com/wiotp/monitoring-dashboard/issues/1152) for reference

image

### PR DESCRIPTION
**Pull request by bjornalm**
Closes #1786

**Summary**
Created a modal for selecting an image.

**Change List (commits, features, bugs, etc)**
- Based the new modal on the ComposableModal component. The width is by default 100% and height is 80%, but can be modified to any value using scss.
- Searching and filtering is done by the component on the data in the content prop
- Searching defaults to the src property but can be modified via the `searchProperty` to be any of the image data properties, e.g. title.
- The image label will be the image file name extracted by the path in the src unless the title property has a value.
- A new ImageTile component based on the Tile component was created to handle both grid and list displays and selection of the images. The tiles will wrap depending on the width of the modal.
- The data of the selected image is passed with the onSubmit callback
- Fixed RTL issue for icon switch icon switch
- Added stories
- Added unit tests

**Acceptance Test (how to verify the PR)**
1. Got to https://deploy-preview-1836--carbon-addons-iot-react.netlify.app?path=/story/watson-iot-imagegallerymodal--basic
2. Test the described functionality

1. Go to https://deploy-preview-1836--carbon-addons-iot-react.netlify.app?path=/story/watson-iot-imagegallerymodal--with-i-18-n
2. Test that i18n can be modified

### CODE REVIEW

File: src/components/ImageGalleryModal/ImageGalleryModal.jsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
```
Comment (davidicus) [inline src/components/ImageGalleryModal/ImageGalleryModal.jsx:3]: I thought we had a lint rule about using `classnames` for this package on import.

File: src/components/ImageGalleryModal/_image-tile.scss [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  border-top: 1px solid $ui-03;
+
+  &:nth-of-type(1) {
+    border-color: transparent;
+  }
+  // The border must be hidded when focused or selected, also for sibling
+  &.#{$prefix}--tile--is-selected,
+  &:focus {
+    border-color: transparent;
+    & + input + .bx--tile {
```
Comment (davidicus) [inline src/components/ImageGalleryModal/_image-tile.scss:47]:
```suggestion
    & + input + .#{$prefix}--tile {
```

just a missing prefix

**Review by davidicus — APPROVED** [review]
Great job.There are just a few minor comments.

**Comment by scottdickerson** [comment]
minor point, but we should probably clear the current selection on searching

**Review by scottdickerson — APPROVED** [review]
LGTM

File: src/components/ImageGalleryModal/ImageGalleryModal.jsx (hunk 1 above)
Comment (bjornalm) [reply src/components/ImageGalleryModal/ImageGalleryModal.jsx:3]: Hmm, there was no linting error for me at least. But I will change into `classnames`.

File: src/components/ImageGalleryModal/_image-tile.scss (hunk 2 above)
Comment (bjornalm) [reply src/components/ImageGalleryModal/_image-tile.scss:47]: Will fix.

File: src/components/ImageGalleryModal/ImageGalleryModal.jsx (hunk 1 above)
Comment (tay1orjones) [reply src/components/ImageGalleryModal/ImageGalleryModal.jsx]: Yeah, the eslint rule isn't in yet but we would like to standardize on the fully lowercase name. https://github.com/carbon-design-system/carbon-addons-iot-react/issues/1325

**Pull request merged**